apps/src/weblab/WebLab.js

```diff
@@ -602,24 +560,62 @@ WebLab.prototype.getAppReducers = function() {
   return reducers;
 };
 
+WebLab.prototype.redux = function() {
+  return {
+    getStore,
+    reducers,
+    actions
+  };
+};
+
+WebLab.prototype.syncBrambleFiles = function(callback = () => {}) {
+  this.brambleHost?.syncFiles(
+    this.getCurrentFileEntries(),
+    this.getCurrentFilesVersionId(),
+    callback
+  );
+};
+
 /**
- * Expose New Relic page action hook for use by the enclosed Bramble application.
+ * Make brambleHost available for file sync operations as soon as it's initialized far
+ * enough to start building its UI. This makes operations like "Start Over" available
+ * even in cases where the mount operation throws an uncaught exception (like the missing
+ * index.html error).
+ * @param {Object} brambleHost
  */
-WebLab.prototype.addPageAction = function(...args) {
-  logToCloud.addPageAction(...args);
+WebLab.prototype.onBrambleMountable = function(brambleHost) {
+  this.brambleHost = brambleHost;
 };
 
-WebLab.prototype.PageAction = makeEnum(
-  logToCloud.PageAction.BrambleError,
-  logToCloud.PageAction.BrambleFilesystemResetSuccess,
-  logToCloud.PageAction.BrambleFilesystemResetFailed
-);
```

Comment:
functionality has moved to CdoBramble:
https://github.com/code-dot-org/code-dot-org/blob/b1d7492b5eb13541022bcb5a5e4f0412da763910/apps/src/weblab/CdoBramble.js#L7-L11